### PR TITLE
Type removal breaking changes

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
This PR contains the first set of breaking changes for 2.0 related to OpenSearch 2.0 - https://github.com/opensearch-project/OpenSearch/issues/2480

The following changes are included in this PR:

1. A CI workflow `integration-unreleased.yml` which runs integ tests for the client against unreleased OpenSearch. This will help in future development to prepare clients with the latest unreleased OpenSearch versions. Currently this workflow will test against 2.0 OpenSearch.
2. The CI workflow `integration.yml` is updated to work for released versions which the client is compatible with.
3. Changes to `.ci/run-tests` to support testing against both released and unreleased OpenSearch versions.
4. Type mapping removal across the APIs and related test changes.
5. Remove indices.exists_type and indices.flush_synced and related files.
 
### Issues Resolved
Part of #145 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .ci/run-opensearch.sh [hunk 1]
```diff
@@ -58,40 +58,62 @@ END
...
+      --ulimit memlock=-1:-1 \
+      --detach="$local_detach" \
+      --health-cmd="$(echo $healthcmd)" \
+      --health-interval=2s \
+      --health-retries=20 \
+      --health-timeout=2s \
+      --rm \
+      -d \
+      opensearch:test;
+  fi
```
Comment (VijayanB) [inline .ci/run-opensearch.sh:115]:
```suggestion
CLUSTER_TAG = $CLUSTER
  if [[ "$IS_UNRELEASED" == "false" ]]; then
    CLUSTER_TAG = $CLUSTER_TAG-secure-$SECURE_INTEGRATION
    echo -e "\033[34;1mINFO: building $CLUSTER container\033[0m"
    docker build \
      --file=.ci/$CLUSTER/Dockerfile \
      --build-arg SECURE_INTEGRATION=$SECURE_INTEGRATION \
      --build-arg OPENSEARCH_VERSION=$OPENSEARCH_VERSION \
      --tag=$CLUSTER_TAG \
      .
 else
   CLUSTER_TAG = $CLUSTER_TAG:test
 fi
 echo -e "\033[34;1mINFO:\033[0m Starting container $node_name \033[0m"

  docker run \
    --name "$node_name" \
    --network "$network_name" \
    --env "ES_JAVA_OPTS=-Xms1g -Xmx1g" \
    "${environment[@]}" \
    "${volumes[@]}" \
    "${security[@]}" \
    --publish "$http_port":9200 \
    --ulimit nofile=65536:65536 \
    --ulimit memlock=-1:-1 \
    --detach="$local_detach" \
    --health-cmd="$(echo $healthcmd)" \
    --health-interval=2s \
    --health-retries=20 \
    --health-timeout=2s \
    --rm \
    -d \
    $CLUSTER_TAG;
```

File: .ci/run-tests [hunk 2]
```diff
@@ -15,6 +15,12 @@ else
     export OPENSEARCH_URL_EXTENSION="http"
 fi
 
+export IS_UNRELEASED=false
+if [[ "$OPENSEARCH_VERSION" == *"SNAPSHOT" ]]; then
+  IS_UNRELEASED=true
+fi
+
+echo -e "\033[1m>>>>> Unreleased is $IS_UNRELEASED >>>>>>>>>>>>>>>>>>>>>>>>>>>>>\033[0m"
```
Comment (VijayanB) [inline .ci/run-tests:23]:
```suggestion
echo -e "\033[1m>>>>> Cluster Version is $OPENSEARCH_VERSION >>>>>>>>>>>>>>>>>>>>>>>>>>>>>\033[0m"
```

File: .ci/run-tests [hunk 3]
```diff
@@ -15,6 +15,12 @@ else
     export OPENSEARCH_URL_EXTENSION="http"
 fi
 
+export IS_UNRELEASED=false
+if [[ "$OPENSEARCH_VERSION" == *"SNAPSHOT" ]]; then
+  IS_UNRELEASED=true
```
Comment (VijayanB) [inline .ci/run-tests:20]: Where is this used?

File: .ci/run-opensearch.sh (hunk 1 above)
Comment (VachaShah) [reply .ci/run-opensearch.sh:115]: For unreleased OpenSearch, the docker image gets built when the `./gradlew assemble` runs for OpenSearch, it is built with tag `opensearch:test`. So when unreleased is true, we only need to do the docker run, docker build happens in the CI workflow. When unreleased is false, we get build the docker image using Dockerfile and then run it.

File: .ci/run-tests (hunk 3 above)
Comment (VachaShah) [reply .ci/run-tests:20]: It is used in `run-opensearch` script to decide whether to get the image from docker, build and run it or use the image from unreleased OpenSearch assemble and run it.

File: .ci/run-opensearch.sh (hunk 1 above)
Comment (VachaShah) [reply .ci/run-opensearch.sh:115]: Ah I misunderstood the refactor, this makes sense! I will make the change.

**Review by VijayanB — APPROVED** [review]

**Review by Shivamdhar — APPROVED** [review]

**Pull request merged**